examples/web/components/CodeSnippetPreview.tsx

```diff
@@ -87,6 +88,63 @@ function getProcessor(lang) {
     );
 }
 
+class Fetcher<K> {
+  data: null | K;
+  promise: null | Promise<void>;
+  error: any;
+  constructor(private fetcher: () => Promise<K>) {}
+  fetch() {
+    if (this.data != null) return this.data;
+    if (this.error != null) throw this.error;
+    if (!this.promise) {
+      this.promise = this.fetcher()
+        .then((data) => {
+          this.data = data;
+        })
+        .catch((error) => {
+          this.error = error;
+        });
+    }
+    throw this.promise;
+  }
+}
+
+const processorStore = new Map<string, Fetcher<ReturnType<typeof remark>>>();
+const getProcessor = (lang: string) => {
+  if (!processorStore.has(lang)) {
+    processorStore.set(lang, new Fetcher(() => fetchProcessor(lang)));
+  }
+  return processorStore.get(lang).fetch();
+};
```

Comment:
This is a simple implementation of data fetcher. 
For more, refer to [How lazy() works internally in React?](https://jser.dev/2023-07-25-how-lazy-works-internally/), or [How Suspense works internally in Concurrent Mode](https://jser.dev/react/2022/04/02/suspense-in-concurrent-mode-1-reconciling/)